### PR TITLE
Add data schema validation in json_schema

### PR DESCRIPTION
**Pull request by touilleMan**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -183,7 +183,17 @@ repos:
...
       - id: check-jsonschema
         name: Validate protocol files with jsonschema
-        files: ^oxidation/libparsec/crates/protocol/schema/
         types_or: [ json5 ]
+        files: ^oxidation/libparsec/crates/protocol/schema/
         args: [ --schemafile, json_schema/protocol.schema.json ]
         additional_dependencies: [ json5 ]
+
+      - id: check-jsonschema
+        name: Validate data files with jsonschema
```
Comment (FirelightFlagboy) [inline .pre-commit-config.yaml:192]:
```suggestion
      - id: check-jsonschema
        name: Validate data files with jsonschema
        alias: check-data-schema
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -183,7 +183,17 @@ repos:
     hooks:
       - id: check-jsonschema
         name: Validate protocol files with jsonschema
```
Comment (FirelightFlagboy) [inline .pre-commit-config.yaml:185]:
```suggestion
      - id: check-jsonschema
        name: Validate protocol files with jsonschema
        alias: check-protocol-schema
```

**Review by FirelightFlagboy — APPROVED** [review]
LGTM

Could you add alias to the `pre-commit` config for the schema check ?

That would simplify using `pre-commit` for only checking `data` or `protocol` schema when working on them

File: .pre-commit-config.yaml (hunk 2 above)
Comment (touilleMan) [reply .pre-commit-config.yaml:185]: done.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (touilleMan) [reply .pre-commit-config.yaml:193]: done.

**Pull request merged**